benchmarks/roman_pots/Snakefile

```diff
@@ -0,0 +1,229 @@
+           detector_config=DETECTOR_CONFIG,
+           num_epochs=MODEL_PY["num_epochs"],
+           learning_rate=MODEL_PY["learning_rate"],
+           size_input=MODEL_PY["size_input"],
+           size_output=MODEL_PY["size_output"],
+           n_layers=MODEL_PY["n_layers"],
+           size_first_hidden_layer=MODEL_PY["size_first_hidden_layer"],
+           multiplier=MODEL_PY["multiplier"],
+           leak_rate=MODEL_PY["leak_rate"]
+           ),
+    expand("results/"+DETECTOR_VERSION+"/{detector_config}/detector_benchmarks/"+SUBSYSTEM+"/"+BENCHMARK+"/artifacts/model_px/num_epochs_{num_epochs}/learning_rate_{learning_rate}/size_input_{size_input}/size_output_{size_output}/n_layers_{n_layers}/size_first_hidden_layer_{size_first_hidden_layer}/multiplier_{multiplier}/leak_rate_{leak_rate}/model_px.pt",
+           detector_config=DETECTOR_CONFIG,
+           num_epochs=MODEL_PX["num_epochs"],
+           learning_rate=MODEL_PX["learning_rate"],
+           size_input=MODEL_PX["size_input"],
+           size_output=MODEL_PX["size_output"],
+           n_layers=MODEL_PX["n_layers"],
+           size_first_hidden_layer=MODEL_PX["size_first_hidden_layer"],
+           multiplier=MODEL_PX["multiplier"],
+           leak_rate=MODEL_PX["leak_rate"]
+           ),
+    expand("results/"+DETECTOR_VERSION+"/{detector_config}/detector_benchmarks/"+SUBSYSTEM+"/"+BENCHMARK+"/artifacts/model_px/num_epochs_{num_epochs}/learning_rate_{learning_rate}/size_input_{size_input}/size_output_{size_output}/n_layers_{n_layers}/size_first_hidden_layer_{size_first_hidden_layer}/multiplier_{multiplier}/leak_rate_{leak_rate}/LossVsEpoch_model_px.png",
+           detector_config=DETECTOR_CONFIG,
+           num_epochs=MODEL_PX["num_epochs"],
+           learning_rate=MODEL_PX["learning_rate"],
+           size_input=MODEL_PX["size_input"],
+           size_output=MODEL_PX["size_output"],
+           n_layers=MODEL_PX["n_layers"],
+           size_first_hidden_layer=MODEL_PX["size_first_hidden_layer"],
+           multiplier=MODEL_PX["multiplier"],
+           leak_rate=MODEL_PX["leak_rate"]
+           )
+ 
+
+
+
+rule roman_pots_generate_events:
+    input:
+        script="steering_file.py"
+    params:
+        detector_path=DETECTOR_PATH,
+        nevents_per_file=NEVENTS_PER_FILE,
+        detector_config=DETECTOR_CONFIG
+    output:
+        "results/"+DETECTOR_VERSION+"/"+DETECTOR_CONFIG+"/detector_benchmarks/"+SUBSYSTEM+"/"+BENCHMARK+"/raw_data/"+DETECTOR_VERSION+"_"+DETECTOR_CONFIG+"_{index}.edm4hep.root"
+    shell:
+      """
+      npsim --steeringFile {input.script} \
+            --compactFile {params.detector_path}/{params.detector_config}.xml \
```

Comment:
I would drop global variables in the leaf rules, and use wildcards where possible
```suggestion
        detector_path=DETECTOR_PATH,
        nevents_per_file=NEVENTS_PER_FILE,
    output:
        "results/"+DETECTOR_VERSION+"/{DETECTOR_CONFIG}/detector_benchmarks/"+SUBSYSTEM+"/"+BENCHMARK+"/raw_data/"+DETECTOR_VERSION+"_{DETECTOR_CONFIG}_{index}.edm4hep.root"
    shell:
      """
      npsim --steeringFile {input.script} \
            --compactFile {params.detector_path}/{wildcards.DETECTOR_CONFIG}.xml \
```